psclient/src/main/java/com/majeur/psclient/service/ShowdownService.kt

```diff
@@ -39,6 +39,21 @@ class ShowdownService : Service() {
     private var previousChatRoomId: String? = null
     private var previousBattleRoomId: String? = null
 
+    val replayManager by lazy {
+        ReplayManager(okHttpClient, object : ReplayManager.ReplayCallback {
+            override fun init() {
+                battleMessageObserver.onSetBattleType(BattleRoomMessageObserver.BattleType.REPLAY)
+            }
+            override fun onMessage(msg: String) {
+                processServerData(msg)
+            }
+
+            override fun onAction(replayAction: BattleRoomMessageObserver.ReplayAction) {
+                battleMessageObserver.handleReplayAction(replayAction)
+            }
+        })
```

Comment:
Instead of the ReplayCallback interface, you can pass directly ShodownService.this and retrieve everything you need inside ReplayManager directly from service reference (such as okHttpClient, battleRoomMessageObserver and processServerData method)

Reply:
Thanks. I think at the time I was just trying to avoid too much coupling to ShowdownService itself, but I think it just made it a bit too complicated for what it was.

I have updated according to your suggestion, and it's much cleaner now.